flickit-assessment-kit/src/main/java/org/flickit/assessment/kit/adapter/out/persistence/kitversion/KitVersionMapper.java

```diff
@@ -35,4 +36,8 @@ public static KitVersion mapToDomainModel(KitVersionJpaEntity entity) {
             entity.getCreatedBy(),
             entity.getLastModifiedBy());
     }
+
+    public static long mapKitVersionToVersionStatus(KitVersionStatus status, long id){
+        return status == KitVersionStatus.ARCHIVE ? -id : status.getId();
```

Comment:
**Suggestion:** Modify the mapKitVersionToVersionStatus method to handle potential null values or unexpected statuses to prevent runtime exceptions. [possible issue, importance: 7]
```suggestion
    public static long mapKitVersionToVersionStatus(KitVersionStatus status, long id){
        if (status == null) return -1;
        return status == KitVersionStatus.ARCHIVE ? -id : status.getId();
    }
```